### PR TITLE
fix: App crash on binding large base64 data to text widget

### LINKED ISSUE #10379: [Bug]: Filepicker crashes the app when we are referencing base64 type files in other widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I add a file in filepicker which is large (>2MB) and set the datatype as base64. My app is crashes when I bind the data with a text widget.

### Steps To Reproduce

1. Go to any app
2. Add a filepicker and set the data format to base64
3. Add a text widget and bind it to filepicker.files
4. Add a file in the filepicker greater than 2mb but less than 5mb. (Note 2mb is a guess, can happen for lower values but not very low values)

Sample file used - [generated_2.txt](https://github.com/appsmithorg/appsmith/files/7860844/generated_2.txt)

Loom link -
[LOOM DEMO](https://www.loom.com/share/55d141477a4240ccb3adeb4f18971487)

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by prsidhu**
## Description

TextWidget:
if text.length > 2000 and number of line breaks in text == 0;
then render only a substring (2000) of the text and display appropriate warning.

Fixes #10379 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

 - Follow the steps in the linked issue description.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by prsidhu** [comment]
Screen Shot 2022-06-23 at 10 58 03 PM

File: app/client/src/workers/validations.ts [hunk 1]
```diff
@@ -264,6 +264,14 @@ function validateExcessLineBreaks(value: any): boolean {
   return lineBreakCount > MAX_ALLOWED_LINE_BREAKS;
 }
 
+function validateExcessLength(text: string): boolean {
```
Comment (SatishGandham) [inline app/client/src/workers/validations.ts:267]: Take max length as a parameter to the function.

File: app/client/src/workers/validations.ts [hunk 2]
```diff
@@ -264,6 +264,14 @@ function validateExcessLineBreaks(value: any): boolean {
...
+  /**
+   * Check if text is too long and without any line breaks.
+   */
+  const lineBreakCount = countOccurrences(text, "\n", false, 0);
```
Comment (SatishGandham) [inline app/client/src/workers/validations.ts:271]: How did we arrive at 20k and 2k?
I can understand giving 10-20% for edge cases, but here we are showing 20k characters, and reducing it to 2k if the length is 20k+

**Review by SatishGandham — CHANGES_REQUESTED** [review]

File: app/client/src/workers/validations.ts (hunk 2 above)
Comment (prsidhu) [reply app/client/src/workers/validations.ts:271]: @SatishGandham no that's a typo error. I'll fix it. The number of characters is a hunch. The problem is that experiment listed in the issue tries to render a single sentence of 5.5 million characters, which works on my system but crashes on Aparna's. So I have tried to play safe with the sub string length (2000). What do you suggest?

**Review by SatishGandham — DISMISSED** [review]

**Comment by chandannkumar** [comment]
@prsidhu Facing app crash(page unresponsive) when we try to bind Filepicker with Text widget after uploading a file on filepicker
https://www.loom.com/share/0f8021588d724b0e86481aaaf9ee6d3f

P.S: It works fine when we bind widgets before uploading a file on filepicker

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**